src/controllers/portfolio/portfolio.test.ts

```diff
@@ -614,6 +615,88 @@ describe('Portfolio Controller ', () => {
     })
   })
 
+  test('To be learned token is returned from portfolio and not passed to learnedTokens (without balance)', async () => {
+    const { storage, controller } = prepareTest()
+    const ethereum = networks.find((network) => network.id === 'ethereum')!
+    const clonedEthereum = structuredClone(ethereum)
+    clonedEthereum.hasRelayer = false
+  
+    await controller.addTokensToBeLearned(
+      ['0xA0b73E1Ff0B80914AB6fe0444E65848C4C34450b'],
+      'ethereum'
+    )
+
+    await controller.updateSelectedAccount(account.addr, clonedEthereum, undefined, {
+      forceUpdate: true,
+    })
+
+    const toBeLearnedToken = controller.latest[account.addr].ethereum?.result?.tokens.find(
+      (token) => token.address === '0xA0b73E1Ff0B80914AB6fe0444E65848C4C34450b'
+    )
+
+    expect(toBeLearnedToken).toBeTruthy()
+
+    const previousHintsStorage = await storage.get('previousHints', {})
+    const tokenIsNotInLearnedTokens =
+      previousHintsStorage.learnedTokens?.ethereum && previousHintsStorage.learnedTokens?.ethereum[toBeLearnedToken!.address]
+
+    expect(tokenIsNotInLearnedTokens).toBeFalsy()
```

Comment:
this name is extremely confusing. `tokenInLearnedTokens`, like it is below is better